src/software/jetson_nano/services/network/network.cpp

```diff
@@ -74,3 +77,61 @@ void NetworkService::primitiveSetCallback(TbotsProto::PrimitiveSet input)
                      << "%";
     }
 }
+
+void NetworkService::logNewPrimitiveSet(const TbotsProto::PrimitiveSet& new_primitive_set)
+{
+    if (primitive_set_rtt.size() >= PRIMITIVE_DEQUE_MAX_SIZE)
+    {
+        LOG(WARNING)
+            << "Too many primitive sets logged for round-trip calculations, halting log process";
+        return;
+    }
+
+    if (!primitive_set_rtt.empty() && new_primitive_set.sequence_number() <=
+                                          primitive_set_rtt.back().primitive_sequence_num)
+    {
+        // If the proto is older than the last received proto, then ignore it
+        return;
+    }
+
+    NetworkService::RoundTripTime current_round_trip_time;
+    current_round_trip_time.primitive_sequence_num = new_primitive_set.sequence_number();
+    current_round_trip_time.thunderscope_sent_time_seconds =
+        new_primitive_set.time_sent().epoch_timestamp_seconds();
+    current_round_trip_time.thunderloop_recieved_time_seconds =
+        getCurrentEpochTimeInSeconds();
```

Comment:
Easier to read using brace initialization
```suggestion
    NetworkService::RoundTripTime current_round_trip_time{
        .primitive_sequence_num = new_primitive_set.sequence_number(),
        .thunderscope_sent_time_seconds = new_primitive_set.time_sent().epoch_timestamp_seconds(),
        .thunderloop_recieved_time_seconds = getCurrentEpochTimeInSeconds()};
```